File: HelloWorld/Swap.java
public class SwapNumbers {

	public static void main(String[] args) {
		int number1 = 10;//binary = 1010
		int number2 = 15; //binary = 1111
		System.out.println("Number1 before swapping:"+number1);
		System.out.println("Number2 before swapping:"+number2);

		number1 = number1^number2; //number1 = 0101
		number2 = number1^number2; //number2 = 1010
		number1 = number1^number2; //number1 = 1111
		
		System.out.println("Number1 after swapping:"+number1);
        System.out.println("Number2 after swapping:"+number2);
        
	}

}

File: HelloWorld/swap.cpp
#include <iostream>
using namespace std;

int main()
{
    int a = 78, b = 45, temp;

    cout << "Before swapping." << endl;
    cout << "a = " << a << ", b = " << b << endl;

    temp = a;
    a = b;
    b = temp;

    cout << "\nAfter swapping." << endl;
    cout << "a = " << a << ", b = " << b << endl;

    return 0;
}